### PR TITLE
{base}[foss/2022a] GFF3-toolkit v2.1.0

### PR DESCRIPTION
**Pull request by ItIsI-Orient**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/g/gff3tool/gff3tool-2.1.0-foss-2022a.eb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+dependencies = [
+    ('Python', '3.10.4'),
+    ('BLAST+', '2.13.0'),
+]
+
+use_pip = True
+sanity_pip_check = True
+download_dep_fail = True
+
+moduleclass = "tools"
```
Comment (verdurin) [inline easybuild/easyconfigs/g/gff3tool/gff3tool-2.1.0-foss-2022a.eb:31]: Shouldn't this be `bio`?

**Review by verdurin — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/g/gff3tool/gff3tool-2.1.0-foss-2022a.eb [hunk 2]
```diff
@@ -0,0 +1,31 @@
+easyblock = 'PythonPackage'
+
+name = 'gff3tool'
```
Comment (boegel) [inline easybuild/easyconfigs/g/gff3tool/gff3tool-2.1.0-foss-2022a.eb:3]: `GFF3-toolkit` makes more sense as software name, based on https://gff3toolkit.readthedocs.io

File: easybuild/easyconfigs/g/gff3tool/gff3tool-2.1.0-foss-2022a.eb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+checksums = [
+    {'gff3tool-2.1.0.tar.gz': 'e842ef4afdb66926a2b63bf2619c7ef30bd9b76a5f1084a5c57bf8b39cc4f15d'},
+    {'GFF3toolkit-2.1.0-remove_hardcode.patch': '03696635af25ab1e3d650b46c7aeaaaadb31e7eac766039c758150f3f3162cb7'},
+]
+
+local_blast_path = '%(installdir)s/lib/python%(pyshortver)s/site-packages/gff3tool/lib/ncbi-blast+'
+preinstallopts = 'mkdir -p %s && ' % local_blast_path
+preinstallopts += 'ln -s $EBROOTBLASTPLUS/bin  %s && ' % local_blast_path
+
+dependencies = [
```
Comment (boegel) [inline easybuild/easyconfigs/g/gff3tool/gff3tool-2.1.0-foss-2022a.eb:22]: Move this up, below `checksums`, above `local_blast_path`

File: easybuild/easyconfigs/g/gff3tool/gff3tool-2.1.0-foss-2022a.eb (hunk 1 above)
Comment (boegel) [reply easybuild/easyconfigs/g/gff3tool/gff3tool-2.1.0-foss-2022a.eb:31]: +1, should be `bio`

File: easybuild/easyconfigs/g/gff3tool/gff3tool-2.1.0-foss-2022a.eb [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+toolchain = {'name': 'foss', 'version': '2022a'}
+
+sources = ['gff3tool-2.1.0.tar.gz']
+patches = ['GFF3toolkit-2.1.0-remove_hardcode.patch']
+checksums = [
+    {'gff3tool-2.1.0.tar.gz': 'e842ef4afdb66926a2b63bf2619c7ef30bd9b76a5f1084a5c57bf8b39cc4f15d'},
+    {'GFF3toolkit-2.1.0-remove_hardcode.patch': '03696635af25ab1e3d650b46c7aeaaaadb31e7eac766039c758150f3f3162cb7'},
+]
+
+local_blast_path = '%(installdir)s/lib/python%(pyshortver)s/site-packages/gff3tool/lib/ncbi-blast+'
```
Comment (boegel) [inline easybuild/easyconfigs/g/gff3tool/gff3tool-2.1.0-foss-2022a.eb:18]: maybe add a line of comment here to explain why this is needed?

File: easybuild/easyconfigs/g/gff3tool/gff3tool-2.1.0-foss-2022a.eb [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+name = 'gff3tool'
+version = '2.1.0'
+
+homepage = 'https://gff3toolkit.readthedocs.io'
+description = 'Python programs for processing GFF3 files'
+
+toolchain = {'name': 'foss', 'version': '2022a'}
+
+sources = ['gff3tool-2.1.0.tar.gz']
+patches = ['GFF3toolkit-2.1.0-remove_hardcode.patch']
```
Comment (boegel) [inline easybuild/easyconfigs/g/gff3tool/gff3tool-2.1.0-foss-2022a.eb:12]: preferred patch filename: `GFF3-toolkit-2.1.0_remove-hardcode.patch`

File: easybuild/easyconfigs/g/gff3tool/gff3tool-2.1.0-foss-2022a.eb [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+name = 'gff3tool'
+version = '2.1.0'
+
+homepage = 'https://gff3toolkit.readthedocs.io'
+description = 'Python programs for processing GFF3 files'
+
+toolchain = {'name': 'foss', 'version': '2022a'}
+
+sources = ['gff3tool-2.1.0.tar.gz']
```
Comment (boegel) [inline easybuild/easyconfigs/g/gff3tool/gff3tool-2.1.0-foss-2022a.eb:11]: don't hardcode the version here, use `%(version)s` template:

```suggestion
sources = ['gff3tool-%(version)s.tar.gz']

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/g/gff3tool/gff3tool-2.1.0-foss-2022a.eb (hunk 3 above)
Comment (ItIsI-Orient) [reply easybuild/easyconfigs/g/gff3tool/gff3tool-2.1.0-foss-2022a.eb:22]: `local_blast_path` is defined for `preinstallopts` so I believe it should stay above them. Having `preinstallopts` above would result in parsing error in Easybuild.

**Comment by ItIsI-Orient** [comment]
@boegelbot please test @ generoso

**Review by boegel — APPROVED** [review]

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3145.skitty.os - Linux RHEL 8.8, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/boegel/abdfc371ca29c684044893ef16f490f9 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @ItIsI-Orient!

**Pull request merged**